bin/cron.js

```diff
@@ -38,6 +38,252 @@ cron.schedule("0 0 0 * * *", () => {
   });
 });
 
+/**
+ * @description - https://www.npmjs.com/package/node-cron; runs every Monday to update every course's badges related to Top scores and Earliest Time Completed
+ */
+cron.schedule("0 0 * * Mon", async () => {
+  const logs = {};
+  Object.keys(config.mongoDBs).map(async (courseID) => {
+    console.log(`Updating ${config.mongoDBs[courseID]}'s user progress`);
+    try {
+      const assignmentIdToType = {}, // Maps a course's modules assignment id to its type - e.g 22657: "practice"
+        badgeIdToPoints = {}; // Maps a course's badges id to its points - e.g 1: 200
+
+      const db = mongo.client.db(config.mongoDBs[courseID]),
+        userSubmissionsPromise = () =>
+          canvas.getSubmissions(
+            courseID,
+            "student_ids[]=all&workflow_state=graded&grouped=true&per_page=1000" // Get submissions grouped by user - e.g [{user_id:1, submissions: []}, ...]
+          ),
+        modulesPromise = () => db.collection("modules").find().sort({ _id: 1 }).toArray(),
+        badgesPromise = () => db.collection("badges").find().sort({ _id: 1 }).toArray();
+
+      // Retrieve all necessary information at once
+      const [userSubmissions, modules, badges] = await Promise.allSettled([
+        userSubmissionsPromise(),
+        modulesPromise(),
+        badgesPromise(),
+      ]);
+
+      //Intitalizes dicts of scores for practice and quizzes
+      let highestPractice = {};
```

Comment:
In general, use `const` first, then `let`, and never `var`. `const` declarations don't let you change what it references, so you should usually always use it for arrays and objects, cause you typically only change their attributes, not their references. Use `let` for primitive that change, otherwise const.

Reply:
Got it